### PR TITLE
First sketch on p:cast-content-type

### PR DESCRIPTION
**Pull request by xml-project**
This is a first sketch for the revised "p:cast-content-type" (see #16). I tried to include @ndw's proposal to parse and serialize e.g. xml->text and vice versa. Also I try to come up for document property "serialization" which was not covered yet.
I do not expect an approve yet, but propose to discuss this on next editors call.

### CODE REVIEW

File: steps/src/main/xml/steps/cast-content-type.xml [hunk 1]
```diff
@@ -1,133 +1,229 @@
...
+               serialization property.
+            </para>
+         </listitem>
+         <listitem>
+            <para>Casting from an XML media type to a JSON media type converts the
+               XML into JSON. <impl>The precise nature of the conversion from XML to JSON
+                  is <glossterm>implementation–defined</glossterm>.</impl>
+               If the input document is an XML representation of JSON as defined in
+               <biblioref linkend="xpath31-functions"/>,
+               implementations <rfc2119>must</rfc2119> use <function>fn:xml-to-json</function>
```
[the remark this answers is not in the document]
Comment (xml-project) [reply steps/src/main/xml/steps/cast-content-type.xml]: Right. I think we should say, that `fn:parse-json(p:xml-to-json())` is used. Sorry, I did not check this.

File: steps/src/main/xml/steps/cast-content-type.xml [hunk 2]
```diff
@@ -1,133 +1,229 @@
...
+      <itemizedlist>
+         <listitem>
+            <para>Casting from an HTML media type to a XML media type changes
+               “<literal>content-type</literal>” document property and removes any
+               serialization property. </para>
+         </listitem>
+         <listitem>
+            <para>Casting from an HTML media type to another HTML media type
+               changes “<literal>content-type</literal>” document property.</para>
+         </listitem>
```
Comment (ndw) [inline steps/src/main/xml/steps/cast-content-type.xml]: What media types, other than text/html are HTML media types?

File: steps/src/main/xml/steps/cast-content-type.xml [hunk 3]
```diff
@@ -1,133 +1,229 @@
...
+   <para>The <tag>p:cast-content-type</tag> step creates a new document by 
+      changing the media type of its input.</para>
+   
+   <p:declare-step type="p:cast-content-type">
+      <p:input port="source" content-types="*/*"/>
+      <p:output port="result" content-types="*/*"/>
+      <p:option name="content-type" required="true" as="xs:string"/>
+      <p:option name="parameters" as="map(xs:QName,item()*)?"/>
+   </p:declare-step>
+
```
Comment (ndw) [inline steps/src/main/xml/steps/cast-content-type.xml]: Should we note explicitly somewhere that this step does nothing if the content type requested is the same as the content type received?

File: steps/src/main/xml/steps/cast-content-type.xml (hunk 2 above)
Comment (xml-project) [reply steps/src/main/xml/steps/cast-content-type.xml]: > [Definition: The “text/html” and “application/xhtml+xml” media types are HTML media types. ]

File: steps/src/main/xml/steps/cast-content-type.xml (hunk 3 above)
Comment (xml-project) [reply steps/src/main/xml/steps/cast-content-type.xml]: Yes, may be that is a good idea in case people wonder.

**Comment by ndw** [comment]
Agreed. Achim to merge this PR after minor editorial changes.

**Comment by xml-project** [comment]
@ndw Would you mind to check the new first para and give me a go, if everything is all right.

File: steps/src/main/xml/steps/cast-content-type.xml [hunk 4]
```diff
@@ -6,7 +6,9 @@
    <title>p:cast-content-type</title>
    
    <para>The <tag>p:cast-content-type</tag> step creates a new document by 
-      changing the media type of its input.</para>
+      changing the media type of its input. If the value of <option>content-type</option>
```
Comment (ndw) [inline steps/src/main/xml/steps/cast-content-type.xml]: ... If the value of the <option>content-type...

**Comment by ndw** [comment]
Proposed a fix for one grammatical, otherwise it's fine.

**Pull request merged**